# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1709828377 -3600
# Parent  6a0f3b7ac98fea37ed91275d471703661e48c63d
Don't set tooltip on Gnome. It's not supported and sometimes crashes.

diff --git a/mailnews/base/src/nsMessengerUnixIntegration.cpp b/mailnews/base/src/nsMessengerUnixIntegration.cpp
--- a/mailnews/base/src/nsMessengerUnixIntegration.cpp
+++ b/mailnews/base/src/nsMessengerUnixIntegration.cpp
@@ -254,20 +254,25 @@ nsMessengerUnixIntegration::UpdateUnread
     const nsCString& tooltip = NS_ConvertUTF16toUTF8(unreadTooltip);
     // Set both title and tooltip. It appears that some platforms do
     // one but not the other.
     app_indicator_set_title(ci, tooltip.get());
-    app_indicator_set_tooltip_full(ci,
-      NULL /* icon */, tooltip.get() /* title */, NULL /* body */
-    );
+    if (!desktop.EqualsLiteral("gnome")) {
+      // Gnome has no tooltips and this call crashes under Wayland :-(
+      app_indicator_set_tooltip_full(ci,
+        NULL /* icon */, tooltip.get() /* title */, NULL /* body */
+      );
+    }
     app_indicator_set_icon_full(ci, icon_pathN, NULL);
     app_indicator_set_status(ci, APP_INDICATOR_STATUS_ACTIVE);
   } else {
     if (mozilla::Preferences::GetBool("mail.biff.show_tray_icon_always", false) ||
       sHiddenWindows.Length() > 0) {
       app_indicator_set_title(ci, "Betterbird");
-      app_indicator_set_tooltip_full(ci,
-        NULL /* icon */, "Betterbird" /* title */, NULL /* body */
-      );
+      if (!desktop.EqualsLiteral("gnome")) {
+        app_indicator_set_tooltip_full(ci,
+          NULL /* icon */, "Betterbird" /* title */, NULL /* body */
+        );
+      }
       app_indicator_set_icon_full(ci, icon_path0, NULL);
       app_indicator_set_status(ci, APP_INDICATOR_STATUS_ACTIVE);
     } else {
       app_indicator_set_status(ci, APP_INDICATOR_STATUS_PASSIVE);
